lib/MusicBrainz/Server/WebService/Serializer/JSON/LD/Artist.pm

```diff
@@ -11,19 +15,27 @@ extends 'MusicBrainz::Server::WebService::Serializer::JSON::LD';
 with 'MusicBrainz::Server::WebService::Serializer::JSON::LD::Role::GID';
 with 'MusicBrainz::Server::WebService::Serializer::JSON::LD::Role::Name';
 with 'MusicBrainz::Server::WebService::Serializer::JSON::LD::Role::LifeSpan' =>
-    { begin_properties => sub { my $artist = shift; return ($artist->type && $artist->type->name eq 'Person') ? qw( foundingDate birthDate ) : qw( foundingDate ) },
-      end_properties   => sub { my $artist = shift; return ($artist->type && $artist->type->name eq 'Person')  ? qw( dissolutionDate deathDate ) : qw( dissolutionDate ) } };
+    { begin_properties => sub { is_person(shift) ? qw( birthDate ) : qw( foundingDate ) },
+      end_properties   => sub { is_person(shift) ? qw( deathDate ) : qw( dissolutionDate ) } };
 with 'MusicBrainz::Server::WebService::Serializer::JSON::LD::Role::Aliases';
 with 'MusicBrainz::Server::WebService::Serializer::JSON::LD::Role::Area' => { include_birth_death => sub { my $artist = shift; return $artist->type && $artist->type->name eq 'Person' } };
 
+sub is_person {
+    my $artist = shift;
+    return ($artist->type_id && $artist->type_id == $ARTIST_TYPE_PERSON) ? 1 : 0;
+}
+
 around serialize => sub {
     my ($orig, $self, $entity, $inc, $stash, $toplevel) = @_;
     my $ret = $self->$orig($entity, $inc, $stash, $toplevel);
 
-    $ret->{'@type'} = ($entity->type && $entity->type->name eq 'Person') ? ['Person', 'MusicGroup'] : 'MusicGroup';
```

Comment:
This check failed if the types weren't loaded; now it checks `type_id` which should always be there.